pages/[slug].js

```diff
@@ -0,0 +1,131 @@
+import { useRouter } from "next/router.js";
+import Image from "next/image";
+import surfspots from "../lib/surfspots";
+import styled from "styled-components";
+import Link from "next/link";
+import back_button from "../public/images/back_button.png";
+
+export default function SpotDetails() {
+  const router = useRouter();
+  const { slug } = router.query;
+
+  const currentSpot = surfspots.find((spot) => spot.slug === slug);
+
+  if (!currentSpot) {
+    return <p>Not found</p>;
+  }
+
+  return (
+    <>
+      <StyledImage>
+        <Image
+          src={currentSpot.image}
+          alt={`cover image of ${currentSpot.name}`}
+          width={800}
+          height={600}
+        />
+        <StyledTitle>{currentSpot.name}</StyledTitle>
+      </StyledImage>
+      <Link href="/">
+        <StyledBackButton
+          src={back_button}
+          alt="Back_Btn"
+          width={30}
+          height={30}
+        />
+      </Link>
+      <StyledButtonSection>
+        <StyledButton>Infos</StyledButton>
+      </StyledButtonSection>
+      <StyledDescriptionSection>
+        <p>{currentSpot.description}</p>
+      </StyledDescriptionSection>
+      <StyledSpecsSection>
+        <p>
+          <StyledSpan>Surfbare Windrichtungen: </StyledSpan>
+          {currentSpot.winddirection}
+        </p>
+        <p>
+          <StyledSpan>Center/Schule: </StyledSpan>
+          {currentSpot.surfcenter}
+        </p>
+        <p>
+          <StyledSpan>Parken: </StyledSpan>
+          {currentSpot.parking}
+        </p>
+        <p>
+          <StyledSpan>Übernachten: </StyledSpan>
+          {currentSpot.camping}
+        </p>
+      </StyledSpecsSection>
+    </>
+  );
+}
+
+const StyledImage = styled.div`
+  img {
+    width: 100vw;
+    height: auto;
+  }
+  position: relative;
+`;
+
+const StyledTitle = styled.h1`
+  background-color: white;
+  padding: 10px;
+  margin: 0;
+  text-align: center;
+  /* margin-left: 15%;
+  margin-right: 15%; */
```

Comment:
always delete commented code before committing, unless it is explaining your code :)